contest/register_contest.py

```diff
@@ -0,0 +1,48 @@
+'''
+    The MIT License (MIT)
+    Copyright (c) 2014 NTHUOJ team
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all
+    copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+    '''
+from contest.models import Contest
+from contest.models import Contestant
+
+from contest.contest_info import get_contest_or_404
+from contest.contest_info import can_register
+
+from group.models import Group
+
+from utils.log_info import get_logger
+
+logger = get_logger()
+
+def register(contest, user):
```

Comment:
Since we have to check ```can_register``` everytime, I suggest to put it into register function. Then we can remove register_user function.
(You can rename register to register_user for readability)